compiler/rustc_lexer/src/unescape.rs

```diff
@@ -294,16 +302,31 @@ where
             _ => scan_escape(first_char, &mut chars, mode),
         };
         let end = initial_len - chars.as_str().len();
-        callback(start..end, unescaped_char);
+        callback(start..end, unescaped_char.map(Ok));
     }
 
-    fn skip_ascii_whitespace(chars: &mut Chars<'_>) {
+    fn skip_ascii_whitespace<F>(chars: &mut Chars<'_>, _start: usize, _callback: &mut F)
+    where
+        F: FnMut(Range<usize>, Result<Result<char, EscapeWarning>, EscapeError>),
+    {
         let str = chars.as_str();
         let first_non_space = str
             .bytes()
             .position(|b| b != b' ' && b != b'\t' && b != b'\n' && b != b'\r')
             .unwrap_or(str.len());
-        *chars = str[first_non_space..].chars()
+        let tail = &str[first_non_space..];
+        if let Some(c) = tail.chars().nth(0) {
+            // For error reporting, we would like the span to contain the character that was not
+            // skipped.  The +1 is necessary to account for the leading \ that started the escape.
+            let end = _start + first_non_space + c.len_utf8() + 1;
+            if c.is_whitespace() {
+                _callback(
+                    _start..end,
```

Comment:
```suggestion
            let end = start + first_non_space + c.len_utf8() + 1;
            if c.is_whitespace() {
                callback(
                    start..end,
```

Reply:
Thanks, sorry about that, I thought I got a warning because of the shadowing.